src/test/java/edu/hm/hafner/analysis/ListTest.java

```diff
@@ -0,0 +1,73 @@
+package edu.hm.hafner.analysis;
+
+import java.util.List;
+
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+import edu.hm.hafner.analysis.assertions.Assertions;
+
+/**
+ * abstract class for list tests, e.g. {@link ArrayListTest}.
+ *
+ * @author Simon Symhoven
+ */
+
+abstract class ListTest {
+    abstract List<Integer> create(int numberOfInitialElements);
+
+    @Test
+    @DisplayName("list should be empty")
+    void isEmpty() {
+        Assertions.assertThat(this.create(0)).isEmpty();
+    }
+
+    @Test
+    @DisplayName("get element of list should be not null")
+    void getElementShouldBeNotNull() {
+        Assertions.assertThat(this.create(5).get(1)).isNotNull();
+    }
+
+    @Test
+    @DisplayName("get element of list should be null (IndexOutOfBounds)")
+    void getElementOutOfIndex() {
+        try {
+            Assertions.assertThat(this.create(5).get(6)).isNull();
+        }
+        catch (IndexOutOfBoundsException e) {
+            System.out.println("IndexOutOfBoundException catched - Test successfull: " + e.getMessage());
+            Assertions.assertThat(true).isTrue();
+        }
```

Comment:
```suggestion
       assertThatIllegalArgumentException().isThrownBy(() -> this.create(5).get(6));
```
Eigentlich müsste auch das this.create(5) noch raus, das funktioniert ja. Siehe auch: https://github.com/uhafner/codingstyle/blob/master/doc/Testen.md#konventionen-beim-schreiben-von-modultests